plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2308,15 +2310,17 @@ private static boolean isSameOrParent(Path parent, Path child)
         return parent.equals(child);
     }
 
-    private void logCleanupFailure(String format, Object... args)
+    @FormatMethod
+    private void logCleanupFailure(@FormatString String format, Object... args)
     {
         if (throwOnCleanupFailure) {
             throw new RuntimeException(format(format, args));
         }
         log.warn(format, args);
     }
 
-    private void logCleanupFailure(Throwable t, String format, Object... args)
+    @FormatMethod
+    private void logCleanupFailure(Throwable t, @FormatString String format, Object... args)
```

Comment:
Yes, we can skip the `@FormatString` if the first `String` parameter is the format string. I learned about this after adding the annotation, but I thought perhaps it would be better to be more explicit. I can remove it, though, for consistency with Airlift.

Reply:
let's drop.

Reply:
Dropped